Wrappers/Python/cil/processors/FluxNormaliser.py

```diff
@@ -0,0 +1,458 @@
+            plt.show()
+            
+    def _plot_slice_roi(self, angle_index=None, channel_index=None, log=False, ax=111):
+        
+        data = self.get_input()
+        if angle_index is not None and 'angle' in data.dimension_labels:
+            data_slice = data.get_slice(angle=angle_index)
+        else:
+            data_slice = data
+        
+        if 'channel' in data.dimension_labels:
+            data_slice = data_slice.get_slice(channel=channel_index)
+
+        if len(data_slice.shape) != 2:
+            raise ValueError("Data shape not compatible with preview_configuration(), data must have at least two of 'horizontal', 'vertical' and 'angle'")
+
+        extent = [0, data_slice.shape[1], 0, data_slice.shape[0]]
+        if 'angle' in data_slice.dimension_labels:
+            min_angle = data_slice.geometry.angles[0]
+            max_angle = data_slice.geometry.angles[-1]
+            for i, d in enumerate(data_slice.dimension_labels):
+                if d !='angle':
+                    extent[i*2]=min_angle
+                    extent[i*2+1]=max_angle
+
+        ax1 = plt.subplot(ax)
+        if log:
+            im = ax1.imshow(numpy.log(data_slice.array), cmap='gray',aspect='equal', origin='lower', extent=extent)
+            plt.gcf().colorbar(im, ax=ax1)
+        else:
+            im = ax1.imshow(data_slice.array, cmap='gray',aspect='equal', origin='lower', extent=extent)
+            plt.gcf().colorbar(im, ax=ax1)
+
+        h = data_slice.dimension_labels[1]
+        v = data_slice.dimension_labels[0]
+
+        if h == 'angle':
+            h_min = min_angle
+            h_max = max_angle
+        else:
+            h_min = self.roi[h][0]
+            h_max = self.roi[h][1]
+
+        if v == 'angle':
+            v_min = min_angle
+            v_max = max_angle
+        else:
+            v_min = self.roi[v][0]
+            v_max = self.roi[v][1]
+
+        ax1.plot([h_min, h_max],[v_min, v_min],'--r')
+        ax1.plot([h_min, h_max],[v_max, v_max],'--r')
+
+        ax1.plot([h_min, h_min],[v_min, v_max],'--r')
+        ax1.plot([h_max, h_max],[v_min, v_max],'--r')
+        
+        title = 'ROI'
+        if angle_index is not None:
+            title += ' angle = ' + str(data.geometry.angles[angle_index])
+        if channel_index is not None:
+            title += ' channel = ' + str(channel_index)
+        ax1.set_title(title)
+
+        ax1.set_xlabel(h)
+        ax1.set_ylabel(v)
+
+    @property
+    def num_threads(self):
+        '''Get the optional number of threads parameter to use for the accelerated version.
+
+        Defaults to the value set in the CIL multiprocessing module.'''
+        return cil_mp.NUM_THREADS if self._num_threads is None else self._num_threads
+
+    def set_num_threads(self, value):
+        '''Set the optional number of threads parameter to use for the accelerated version.
+
+        This is discarded if ``accelerated=False``.'''
+        self._num_threads = value
```

Comment:
I'm not sure it's worth exposing this or not. in general we set the threads to half the number of cores as we don't check if hyperthreading is on or off. These operations are completely memory limited, not compute so more threads won't help over physical core numbers.